packages/ses/src/error/tame-console.js

```diff
@@ -37,5 +38,34 @@ export const tameConsole = (
     };
   }
   const causalConsole = makeCausalConsole(originalConsole, loggedErrorHandler);
+
+  // Attach platform-specific error traps such that any error that bottoms out
+  // an event will get unwrapped by the causal console, revealing the error
+  // stack to the console for debugging.
+
+  // Node.js
+  if (
+    typeof process === 'object' &&
+    process !== null &&
+    typeof process.on === 'function' &&
+    typeof process.exit === 'function'
+  ) {
+    process.on('uncaughtException', error => {
+      causalConsole.error(error);
+      process.exit(process.exitCode || -1);
```

Comment:
We cannot do that in general. That would enable confined code to cause the node process running it to exit. Causing the node process to exit *must* require some privilege.

Reply:
I believe I’ve addressed this concern, albeit pragmatically, with the `errorTrapping` option. The default is consistent behavior with the underlying platform, which is sometimes inconsistent with this principle. The overrides allow for more principled behavior and we’ll certainly use `report` for Endo daemons.